src/form/bf_mx.jl

```diff
@@ -76,315 +43,748 @@ function variable_mc_branch_series_current_prod_hermitian(pm::AbstractUBFModels;
+q, with appropriate substitutions of the variables and parameters:
+p->q, a->b, α->β, pmin->qmin, pmax->qmax
+"""
+function constraint_pqw(model::JuMP.Model, w, p, a::Real, α::Real, wmin::Real, wmax::Real, pmin::Real, pmax::Real)
+    if a==0
+        JuMP.@constraint(model, p==0)
+    else
+        @assert(α>=0, "α has to greater than or equal to zero.")
 
-    p_fr = _PMs.var(pm, n, :P_mx)[f_idx]
-    q_fr = _PMs.var(pm, n, :Q_mx)[f_idx]
+        # CONSTANT POWER
+        if α==0
+            JuMP.@constraint(model, p==a)
 
-    w_to_re = _PMs.var(pm, n, :W_re)[t_bus]
-    w_fr_re = _PMs.var(pm, n, :W_re)[f_bus]
+        # CONSTANT IMPEDANCE
+        elseif α==2
+            JuMP.@constraint(model, p==a*w)
 
-    w_to_im = _PMs.var(pm, n, :W_im)[t_bus]
-    w_fr_im = _PMs.var(pm, n, :W_im)[f_bus]
+        # CONE INCLUSIONS
+        else
+            # cone inclusions have an affine over/under estimator
+            # boundary
+            if a>0
+                l = (1/a)*(pmax-pmin)/(wmax-wmin)*(w-wmin) + pmin/a
+            else
+                # swap pmin and pmax if a<0, because pmin/a > pmax/a
+                l = (1/a)*(pmin-pmax)/(wmax-wmin)*(w-wmin) + pmax/a
+            end
+            # affine overestimator
+            if α>2
+                JuMP.@constraint(model, p/a <= l)
+            # affine underestimator
+            elseif 0<α<2
+                JuMP.@constraint(model, p/a >= l)
+            end
+
+            # constant current case, simplifies to a RotatedSecondOrderCone
+            if α==1
+                #       p/a <= w^(1/2)
+                # <=>   (p/a)^2 <= w
+                # <=>   2*(w/2)*1 >= ||p/a||^2_2
+                # <=>   (w/2, 1, p/a) ∈ RotatedSecondOrderCone(3)
+                JuMP.@constraint(model, [w/2, 1, p/a] in JuMP.RotatedSecondOrderCone())
+            # general power cone
+            elseif 0<α<2
+                #       p/a <= w^(α/2)
+                # <=>   w^(α/2) >= p/a
+                # <=>   (w, 1, p/a) ∈ PowerCone(3)
+                JuMP.@constraint(model, [w, 1, p/a] in MathOptInterface.PowerCone(α/2))
+            # general power cone
+            else # α>2
+                #       p/a >= w^(α/2)
+                # <=>   (p/a)^(2/α) >= w
+                # <=>   (p/a, 1, w) ∈ PowerCone(3)
+                JuMP.@constraint(model, [p/a, 1, w] in MathOptInterface.PowerCone(2/α))
+            end
+        end
+    end
+end
 
-    ccm_re =  _PMs.var(pm, n, :CC_re)[i]
-    ccm_im =  _PMs.var(pm, n, :CC_im)[i]
 
-    JuMP.@constraint(pm.model, p_fr + p_to .==  w_fr_re*(g_sh_fr)' + w_fr_im*(b_sh_fr)' + r*ccm_re - x*ccm_im +  w_to_re*(g_sh_to)'  + w_to_im*(b_sh_to)')
-    JuMP.@constraint(pm.model, q_fr + q_to .==  w_fr_im*(g_sh_fr)' - w_fr_re*(b_sh_fr)' + x*ccm_re + r*ccm_im +  w_to_im*(g_sh_to)'  - w_to_re*(b_sh_to)')
+"""
+Creates the constraints modelling the (relaxed) voltage-dependent loads.
+"""
+function constraint_mc_load(pm::AbstractUBFModels, load_id::Int; nw::Int=pm.cnw)
```

Comment:
does this function require the matrix formulation? if so the name should reflect it, if not, I think this should go into a different file, maybe `src/form/bf.jl`?

Reply:
This belongs to the SDP BF relaxation that formulates KCL as a **vector** constraint. However, at the load component level it still contains **matrix** variables. Therefore, I believe it is in the right place (see general comment below as well).